manifests/snapshot_repository.pp

```diff
@@ -0,0 +1,104 @@
+#  This define allows you to insert, update or delete Elasticsearch snapshot
+#  repositories.
+#
+# @param ensure
+#   Controls whether the named index template should be present or absent in
+#   the cluster.
+#
+# @param api_basic_auth_password
+#   HTTP basic auth password to use when communicating over the Elasticsearch
+#   API.
+#
+# @param api_basic_auth_username
+#   HTTP basic auth username to use when communicating over the Elasticsearch
+#   API.
+#
+# @param api_ca_file
+#   Path to a CA file which will be used to validate server certs when
+#   communicating with the Elasticsearch API over HTTPS.
+#
+# @param api_ca_path
+#   Path to a directory with CA files which will be used to validate server
+#   certs when communicating with the Elasticsearch API over HTTPS.
+#
+# @param api_host
+#   Host name or IP address of the ES instance to connect to.
+#
+# @param api_port
+#   Port number of the ES instance to connect to
+#
+# @param api_protocol
+#   Protocol that should be used to connect to the Elasticsearch API.
+#
+# @param api_timeout
+#   Timeout period (in seconds) for the Elasticsearch API.
+#
+# @param repository_type
+#   Snapshot repository type.
+#
+# @param compress
+#   Location of snapshots. Mandatory
```

Comment:
Accidentally swapped this with location?

Reply:
Good spot, corrected.